src/main/java/de/tum/in/www1/artemis/service/connectors/bamboo/dto/BambooBuildResultNotificationDTO.java

```diff
@@ -49,6 +55,48 @@ public void setBuild(BambooBuildDTO build) {
         this.build = build;
     }
 
+    @Override
+    public ZonedDateTime getBuildRunDate() {
+        return getBuild().getBuildCompletedDate();
+    }
+
+    @Override
+    public Optional<String> getCommitHashFromAssignmentRepo() {
+        return getCommitHashFromRepo(ASSIGNMENT_REPO_NAME);
+    }
+
+    @Override
+    public Optional<String> getCommitHashFromTestsRepo() {
+        return getCommitHashFromRepo(TEST_REPO_NAME);
+    }
+
+    @Override
+    public boolean isBuildSuccessful() {
+        return getBuild().isSuccessful();
+    }
+
+    @Override
+    public Long getBuildScore() {
+        // the real score is calculated in the grading service
+        return 0L;
+    }
+
+    @Override
+    public String getTestsPassedString() {
+        if (getBuild().getTestSummary().getDescription().equals("No tests found")) {
```

Comment:
Could this potentially lead to a NullPointerException? Codacy also recommends it this way: 
`"No tests found".equals(getBuild().getTestSummary().getDescription())`;